### PR TITLE
Update spring-webmvc to Instrumenter API

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/DispatcherServletInstrumentation.java [hunk 1]
```diff
@@ -84,32 +77,24 @@ public static void onEnter(
...
-    }
-  }
-
-  public static class ErrorHandlerAdvice {
-    @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void nameResource(@Advice.Argument(3) Exception exception) {
-      if (exception != null) {
-        // It is fine to set status=ERROR here, end(Context, Response) call will overwrite it if the
-        // exception turns out to be a valid result
-        tracer().onException(Java8BytecodeBridge.currentContext(), exception);
```
Comment (trask) [inline instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/DispatcherServletInstrumentation.java:111]: this didn't fit the Instrumenters well, but doesn't seem needed after removing `hasExceptionOnServerSpan=true` from the test

File: instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/HandlerAdapterInstrumentation.java [hunk 2]
```diff
@@ -57,14 +57,19 @@ public static void nameResourceAndStartSpan(
         @Advice.Argument(2) Object handler,
         @Advice.Local("otelContext") Context context,
         @Advice.Local("otelScope") Scope scope) {
+      // TODO (trask) should there be a way to customize Instrumenter.shouldStart()?
```
Comment (anuraaga) [inline instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/HandlerAdapterInstrumentation.java:60]: Is this the usual nested span problem and not specific to instrumenter API?

**Review by anuraaga — APPROVED** [review]

File: instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/HandlerAdapterInstrumentation.java (hunk 2 above)
Comment (trask) [reply instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/HandlerAdapterInstrumentation.java:60]: interesting, ya, we could look at it that way, if we introduce a `HandlerSpan` (similar to `ServerSpan` and `ClientSpan`), i'll open an issue to discuss

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/SpringWebMvcInstrumenters.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                GlobalOpenTelemetry.get(), INSTRUMENTATION_NAME, new HandlerSpanNameExtractor())
+            .newInstrumenter(SpanKindExtractor.alwaysInternal());
+
+    MODEL_AND_VIEW_INSTRUMENTER =
+        Instrumenter.<ModelAndView, Void>newBuilder(
+                GlobalOpenTelemetry.get(),
+                INSTRUMENTATION_NAME,
+                new ModelAndViewSpanNameExtractor())
+            .addAttributesExtractor(new ModelAndViewAttributesExtractor())
+            .newInstrumenter(SpanKindExtractor.alwaysInternal());
```
Comment (mateuszrzeszutek) [inline instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/SpringWebMvcInstrumenters.java:32]: Nit: no need to specify INTERNAL, it's the default

```suggestion
            .newInstrumenter();

    MODEL_AND_VIEW_INSTRUMENTER =
        Instrumenter.<ModelAndView, Void>newBuilder(
                GlobalOpenTelemetry.get(),
                INSTRUMENTATION_NAME,
                new ModelAndViewSpanNameExtractor())
            .addAttributesExtractor(new ModelAndViewAttributesExtractor())
            .newInstrumenter();
```

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**